test-fixture/src/lib.rs

```diff
@@ -75,9 +75,13 @@ pub fn handshake(client: &mut Connection, server: &mut Connection) {
     let mut b = server;
     let mut datagram = None;
     let is_done = |c: &Connection| matches!(c.state(), State::Connected | State::Closing { .. } | State::Closed(..));
-    while !is_done(a) {
+    let authentication_needed = |e| matches!(e, ConnectionEvent::AuthenticationNeeded);
+    while !is_done(a) || !is_done(b) {
         let d = a.process(datagram, now());
         datagram = d.dgram();
+        if a.events().any(authentication_needed) {
+            a.authenticated(0, now());
```

Comment:
This is great, but can we add a function to the fixture to conditionally authenticate?  I see a lot of code in the above tests that effectively duplicate this.  It might need to return a boolean so that callers can assert that authentication happened as expected.

One thing to note (perhaps in a comment) is that this is destructive.  It consumes any outstanding events on the connection.  That's not a problem for this function, but it is worth noting because it's a potentially surprising side-effect.